### PR TITLE
MRG: Revise overview 1 "Behavioural Analysis" issue #35 and issue #8

### PR DESCRIPTION
**Pull request by shamindras**
@peterasujan @dylandaniels - First draft

Could you please review?

### CODE REVIEW

File: doc/problem/behavior.md [hunk 1]
```diff
@@ -52,3 +59,40 @@ For In/Active State:
...
+
+## Data Requirements Description
+The data we have:
+- a dataframe of observations of location for each mice, (x, y, t) with \Delta t small.
+
+The data we require:
+- The above dataframe with AS/ IS properly classified at each point t for each mouse
+- The above dataframe with a classification of drinking, feeding and locomotion for each mouse at each time t
+- The above dataframe with a classification of consumption size and moving distance with each event at each time t
+- The above dataframe with a classification of strain number and mouse number at each time t (if available)
```
Comment (dylandaniels) [inline doc/problem/behavior.md]: We definitely already have this, it's the consumption size data we don't know if we have.

**Comment by dylandaniels** [comment]
Looks good! Just had one comment I put in-line.

File: doc/problem/behavior.md [hunk 2]
```diff
@@ -1,31 +1,38 @@
...
+
+![alt tag](http://cenzhuoyao.com/wp-content/uploads/2016/04/project1_behavior_profile.png)
 
-Bouts: split all the time duration to several event bout
-- Bout size, Bout routine: the number and frequency of bouts in one event
-- Bout duration, Bout density: the length of time and frequency of bout 
+The main focus is 3 key states of the mice i.e. [_Drinking | Feeding | Locomotion_]
+
+- [_Drinking | Feeding | Locomotion_]
+    - AS [_Drinking | Feeding | Locomotion_] Intensity
```
Comment (dylandaniels) [inline doc/problem/behavior.md]: A note about intensity: We are not entirely sure what the Tecott Lab's meaning of "intensity" is. Our current hypothesis is that intensity is defined as quantity over active state time. E.g. for drinking, intensity is the quantity consumed divided by the total amount of time the mouse is in an active state.

File: doc/problem/behavior.md [hunk 3]
```diff
@@ -52,3 +59,40 @@ For In/Active State:
...
+
+- Key idea is that if we have the most granular dataframe in `Data Requirements Description` then the python code is really just a SQL (in `pandas` form) filtering/ grouping query to generate the required output metrics (from flowchart) in the form of a list
+
+## Testing Framework Outline
+
+## Additional Remarks
+- It is not clear exactly how the existing metrics are to be calculated in the form a single query
+- Not sure yet whether the required dataframe at the most granular level can be easily constructed. This would be really useful for all projects to use so we should really consider developing it for the wider team.
+- Some of the required data metrics like consumption of food/ water at each time t may not be easy to obtain as they are provided for each interval. These may have to be prorated across each time t in some stable way in the construction of the required dataframe
+- We also believe that the metrics provided at each point are single point statistics i.e. means. We should consider outputing the actual histogram of values at each point for the given metric rather than just the single-valued mean metrics
```
Comment (dylandaniels) [inline doc/problem/behavior.md]: For example, we may not only be interested in the average amount of active time spent in locomotion, but the distribution of locomotion. This is a more complicated metric than those outlined in the work by the Tecott Lab's papers referenced below. With this information, we could potentially see interesting trends: the proportion of a mouse-day spent in locomotion could be the same in two time chunks, but the types of movements (distances) could form a more nuanced distribution.

File: doc/problem/behavior.md [hunk 4]
```diff
@@ -1,31 +1,38 @@
...
-- AS Event Intensity: the average consumption size or distance in one minute of AS
-- Event size, Event rate: the number and frequency of events in AS time
+The above flowchart shows the key metrics that are required by the study to capture the behavioural profile:
+
+![alt tag](http://cenzhuoyao.com/wp-content/uploads/2016/04/project1_behavior_profile.png)
 
-Bouts: split all the time duration to several event bout
-- Bout size, Bout routine: the number and frequency of bouts in one event
-- Bout duration, Bout density: the length of time and frequency of bout 
+The main focus is 3 key states of the mice i.e. [_Drinking | Feeding | Locomotion_]
```
Comment (dylandaniels) [inline doc/problem/behavior.md]: Each of these metrics can be seen visually in the slides referenced below. Each metric is a tree, decomposed into two child node metrics, whereby the when the child nodes are multiplied together, they yield the parent metric.

**Comment by shamindras** [comment]
@dylandaniels - have added your changes.

@peterasujan - could you please do a review. Possibly add some material in the ` Testing Framework Outline` section.

File: doc/problem/behavior.md [hunk 5]
```diff
@@ -1,31 +1,41 @@
...
 
-![alt tag](http://cenzhuoyao.com/wp-content/uploads/2016/04/project1_behavior_profile.png)
+1. how they drink, feed or move (locomotion) and
+2. how they translate between active state or inactive state.
 
-## Background Knowledge:
-- Mouses react differently during Active state and Inactive states and all behavioral record should be classified into 2 mutually exclusive categories, Active States (ASs) and Inactive States (ISs). To designate ISs, we examined all time intervals occurring between movement, feeding, and drinking events while the animal was outside the Home base. Those time intervals exceeding an IS Threshold (IST) duration value were classified as ISs; the set of ASs was then defined as the complement of these ISs. Equivalent mathematically, ASs can also be defined as those intervals resulting from connecting gaps between events outside the Home base of length at most IST; ISs are then defined as the complement of these ASs. (*Active State Organization of Spontaneous Behavior Patterns*, C. Hillar et al.)
+From the paper we have the following key background information from the paper:
```
Comment (peterasujan) [inline doc/problem/behavior.md]: Typo: this sentence doesn't need "from the paper" twice.

File: doc/problem/behavior.md [hunk 6]
```diff
@@ -52,3 +62,41 @@ For In/Active State:
...
+    - a dictionary containing the rectangular vertices marking the area to restrict the movement to i.e. x_lower, x_upper, y_lower, y_upper.
+    - [_Drinking | Feeding | Locomotion_] state specification
+- The main output is a list containing the key metrics stated in `Statement of statistical problems` section
+
+- Key idea is that if we have the most granular dataframe in `Data Requirements Description` then the python code is really just a SQL (in `pandas` form) filtering/ grouping query to generate the required output metrics (from flowchart) in the form of a list
+
+## Testing Framework Outline
+
+## Additional Remarks
+- It is not clear exactly how the existing metrics are to be calculated in the form a single query
```
Comment (peterasujan) [inline doc/problem/behavior.md]: This sentence doesn't totally make sense. Perhaps it should say "in order to form a single query" ?

**Comment by jarrodmillman** [comment]
Please rebase this before merging.

**Pull request merged**